src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java

```diff
@@ -270,11 +275,13 @@ static void assertGraal(ProjectConfiguration configuration) throws IOException {
         String graalPathString = configuration.getGraalPath();
         if (graalPathString == null) throw new IllegalArgumentException("There is no GraalVM in the projectConfiguration");
         Path graalPath = Path.of(graalPathString);
-        if (!graalPath.toFile().exists()) throw new IOException ("Path provided for GraalVM doesn't exist: "+graalPathString);
+        if (!graalPath.toFile().exists()) throw new IOException("Path provided for GraalVM doesn't exist: " + graalPathString);
         Path binPath = graalPath.resolve("bin");
-        if (!binPath.toFile().exists()) throw new IOException("Path provided for GraalVM doesn't contain a bin directory: "+graalPathString);
-        Path niPath = binPath.resolve("native-image");
-        if (!niPath.toFile().exists()) throw new IOException ("Path provided for GraalVM doesn't contain bin/native-image: "+graalPathString);
+        if (!binPath.toFile().exists()) throw new IOException("Path provided for GraalVM doesn't contain a bin directory: " + graalPathString);
+        Path niPath = Constants.OS_WINDOWS.equals(configuration.getHostTriplet().getOs()) ?
+                binPath.resolve("native-image.cmd") :
```

Comment:
Use `getNativeImageCommand()`?

Reply:
This is indeed a little bit tricky without too much refactoring. I propose we tackle this when we clean up the host and target configuration code.

Reply:
Makes sense